profiles/serializers.py

```diff
@@ -284,9 +284,45 @@ class Meta:
 
 
 class SavedCompanySerializer(serializers.ModelSerializer):
+    company_pk = serializers.IntegerField(write_only=True)
+
     class Meta:
         model = SavedCompany
-        fields = ("id", "user", "company", "added_at")
+        fields = ("id", "user", "company", "company_pk", "added_at")
+        read_only_fields = [
+            "user",
+            "company",
+        ]
+
+    def validate(self, attrs):
+        user = self.context["request"].user
+        company_pk = attrs["company_pk"]
+        if not Profile.objects.filter(pk=company_pk).exists():
+            raise serializers.ValidationError(
+                {
+                    "company_pk": [
+                        f'Invalid pk "{company_pk}" - object does not exist.'
```

Comment:
I recommend to make validation error messages 'stable' (not having IDs and other variable stuff included). "Company not found" should be OK. There is no need to duplicate information which we have in request.